src/RequestStatus.cpp

```diff
@@ -3,15 +3,16 @@
 namespace lift
 {
 
-static const std::string REQUEST_STATUS_BUILDING          = "BUILDING";
-static const std::string REQUEST_STATUS_EXECUTING         = "EXECUTING";
-static const std::string REQUEST_STATUS_SUCCESS           = "SUCCESS";
-static const std::string REQUEST_STATUS_CONNECT_ERROR     = "CONNECT_ERROR";
-static const std::string REQUEST_STATUS_CONNECT_DNS_ERROR = "CONNECT_DNS_ERROR";
-static const std::string REQUEST_STATUS_CONNECT_SSL_ERROR = "CONNECT_SSL_ERROR";
-static const std::string REQUEST_STATUS_TIMEOUT           = "TIMEOUT";
-static const std::string REQUEST_STATUS_RESPONSE_EMPTY    = "RESPONSE_EMPTY";
-static const std::string REQUEST_STATUS_ERROR             = "ERROR";
+static const std::string REQUEST_STATUS_BUILDING                    = "BUILDING";
+static const std::string REQUEST_STATUS_EXECUTING                   = "EXECUTING";
+static const std::string REQUEST_STATUS_SUCCESS                     = "SUCCESS";
+static const std::string REQUEST_STATUS_CONNECT_ERROR               = "CONNECT_ERROR";
+static const std::string REQUEST_STATUS_CONNECT_DNS_ERROR           = "CONNECT_DNS_ERROR";
+static const std::string REQUEST_STATUS_CONNECT_SSL_ERROR           = "CONNECT_SSL_ERROR";
+static const std::string REQUEST_STATUS_TIMEOUT                     = "TIMEOUT";
+static const std::string REQUEST_STATUS_RESPONSE_EMPTY              = "RESPONSE_EMPTY";
+static const std::string REQUEST_STATUS_ERROR                       = "ERROR";
+static const std::string REQUEST_STATUS_DOWNLOADED_TOO_MANY_BYTES   = "DOWNLOADED_TOO_MANY_BYTES";
```

Comment:
realistically wouldn't you just chop off the extra bytes? i'm a bit confused by this error

Reply:
I was mostly trying to encapsulate an error which might occur when we write more bytes than the set max bytes -- but maybe it would be better to have this indicate a general error in the write callback function?

Reply:
It might signify an error in your code, but realistically as an end user the only error you would get is a write error when the remote server fails and doesn't send enough bytes -- I just don't really see it as being very useful since you can always 'chop' off extra bytes if you did happen to save too many

Reply:
Okay, I think I'm on the right page now! Thank you for the comments!